src/AutoFactories.Microsoft.DependencyInjection/Views/Partial/FactoryMethod.hbs

```diff
@@ -1,4 +1,4 @@
-﻿            return new {{ReturnType.QualifiedName}}(
+﻿            return new {{Type.QualifiedName}}(
```

Comment:
This was the crux of the problem. I twas using the return type which when used ExposedAs would be the wrong type